File: java-sdk/radar-schemas-core/src/main/java/org/radarbase/schema/specification/active/AppActiveSource.java
package org.radarbase.schema.specification.active;

import static org.radarbase.schema.util.SchemaUtils.expandClass;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSetter;
import org.radarbase.schema.specification.AppDataTopic;

public class AppActiveSource extends ActiveSource<AppDataTopic> {
    @JsonProperty("app_provider")
    private String appProvider;

    @JsonSetter
    @SuppressWarnings("PMD.UnusedPrivateMethod")
    private void setAppProvider(String provider) {
        this.appProvider = expandClass(provider);
    }

    public String getAppProvider() {
        return appProvider;
    }
}

File: java-sdk/radar-schemas-core/src/main/java/org/radarbase/schema/specification/active/AppActiveSource.kt
package org.radarbase.schema.specification.active

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonSetter
import org.radarbase.config.OpenConfig
import org.radarbase.schema.specification.AppDataTopic
import org.radarbase.schema.util.SchemaUtils

@JsonInclude(NON_NULL)
@OpenConfig
class AppActiveSource : ActiveSource<AppDataTopic>() {
    @JsonProperty("app_provider")
    @set:JsonSetter
    var appProvider: String? = null
        set(value) {
            field = SchemaUtils.expandClass(value)
        }
}
